rpc/client.go

```diff
@@ -94,9 +94,10 @@ type Client struct {
 // method is invoked.
 func NewClient(addr net.Addr, opts *retry.Options, context *Context) *Client {
```

Comment:
this has entirely too much logic in it to not return an error.
We should either have NewClient return errors properly, or have just the basic initialization. Connection would be done later.